### PR TITLE
Allow adding of multiple attachments

### LINKED ISSUE #11: Multi file upload
Multiple people have requested that you can upload multiple files / pictures.

### PR DESCRIPTION
**Pull request by Raymonzut**
Fix #11

changes ported from A–Eskwadraat' fork
  ^ > declaratie-formulier@0aa526
  ^ + give to back-end as `tickets` i.o. `ticket`
  ^ + works with mailgun
  
todo:
- [x] Allow selection of multiple files
- [x] Pass the multiple files to mailgun
- [ ] ~~Rewrite `$files` from SoA (Structure of Arrays) to AoS (Array of Structures) if it is as simple as the current implementation~~

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -186,13 +186,21 @@ function validateEmail($email) {
 			&& preg_match('/@.+\./', $email);
 }
 
-// Validates wheter the uploaded file is of
+function validateTickets($files) {
+	$success = true;
+	for($i = 0; $i < count($files['name']); $i++) {
```
Comment (Riscky) [inline index.php:191]: don't you want to count `$files` here, instead of the length of the `name` attribute?

File: index.php [hunk 2]
```diff
@@ -220,7 +228,7 @@ function validateTicket($file) {
 	// A file is valid if it has an allowed extension,
 	// an allowed filetype and is less than or equal
 	// in size to the maximum filesize.
-	if($file['size'] <= FILE_MAX_FILESIZE ){
+	if($file['size'][$index] <= FILE_MAX_FILESIZE ){
```
Comment (Riscky) [inline index.php:231]: I suppose this works, but wouldn't `$file[$index]['size']` be a lot more readable? I guess that would mean a bit of refactor

**Review by Riscky — CHANGES_REQUESTED** [review]
In general this looks good, but I would store a single array of files instead of separate arrays for each attribute of the files. That would make things a bit more readable.

File: index.php (hunk 1 above)
Comment (Raymonzut) [reply index.php:191]: No, that would count the attributes, remember `$files` is SoA here.
See the next comment about the consideration on changing this.

File: index.php (hunk 2 above)
Comment (Raymonzut) [reply index.php:231]: Agreed!
It also got me itching when porting these changes over from the fork, and rewriting files to be AoS is a nice to have for readability if the implementation is as simple.

So I will look into this rewrite soon, will draft this PR for now :D

Comment (Riscky) [reply index.php:231]: I'm looking forward to the next review :)

**Comment by Raymonzut** [comment]
@Riscky
So I scrapped the last todo (the rewrite), here is why:

As said before, it is a nice to have if the implementation is as simple; It would allow ditching the explicit indexes, which fits better with having the data passed down as function arguments.
There isn't such thing as `<input name="ticket[]"`:SoA to `<input name="[]ticket"`:AoS, and converting the `$_POST` is out of the question.

The style of destructuring on top and accessing those is also not better, as we also prefer passing `$_POST` and `$_FILES` explicitly instead of grabbing them from global scope.

Neither does other syntactic sugar allow a better rewrite, nor we should: the code fits the provided input format, and is quite readable.

**Comment by Riscky** [comment]
> So I scrapped the last todo (the rewrite), here is why

Thorough explanation, I'll approve the PR :)

**Review by Riscky — APPROVED** [review]

**Pull request merged**